src/Workspaces/Core/Portable/Workspace/Solution/DocumentInfo.cs

```diff
@@ -184,20 +202,38 @@ internal sealed class DocumentAttributes : IChecksummedObject, IObjectWritable
             /// </summary>
             public bool IsGenerated { get; }
 
+            /// <summary>
+            /// True if the source code contained in the document is only used in design-time (e.g. for completion),
+            /// but is not passed to the compiler when the containing project is built.
+            /// </summary>
+            public bool DesignTimeOnly { get; }
```

Comment:
What worries me about this particular property is it's unclear how it's expected to be used. If I'm a workspace host, and I give it true...what am I signing up for?

Reply:
What's not clear about it? As the comment suggests all documents that are added to the solution that are not part of the compilation should be marked by this flag.